src/utils/engine.js

```diff
@@ -0,0 +1,75 @@
+import uuid from './uuid'
+
+export default class Engine {
+  constructor (callback) {
+    this.computeQueue = []
+    this.computing = new Set()
+    this.computations = {}
+    this.callback = callback
+  }
+
+  _queueNode (node, dependsOn, notDependsOn) {
+    const index = this.computeQueue.findIndex(element => element.node === node)
+    let dependencies = dependsOn ? [dependsOn] : []
+    if (index >= 0) {
+      const previous = this.computeQueue.splice(index, 1)[0]
+      dependencies = previous.dependencies.concat(dependencies)
+    }
+    const deprecatedIndex = dependencies.indexOf(notDependsOn)
+    if (deprecatedIndex >= 0) dependencies.splice(deprecatedIndex, 1)
+    node.computing = true
+    return this.computeQueue.push({ dependencies, node }) - 1
+  }
+
+  queueComputation (start, dependsOn, notDependsOn, lazy = false) {
+    // Traverse canvas breadth-first to determine update order
+    let index = this._queueNode(start, dependsOn, notDependsOn)
+    while (index < this.computeQueue.length) {
+      const node = this.computeQueue[index].node
+      const nextNodes = new Set(node.edges.filter(edge => edge.start.node === node).map(edge => edge.end.node))
+      nextNodes.forEach(next => this._queueNode(next, node))
+      index++
+    }
+    if (!lazy) return this.compute()
```

Comment:
`lazy` allows queuing of multiple nodes before manually starting the computation.